### ISSUE TITLE
Add verify method to secp256k1.PublicKey

### ISSUE THREAD

**Issue by robdefeo** [mailchain/mailchain#345]
**Is your feature request related to a problem? Please describe.**

It is not possible to verify if a `[]byte` is signed by a public key.  This is a problem, when determining the public key from an address and there are multiple different public key possibilities due to the protocol allowing different key types.

**Describe the solution you'd like**

Implement `Verify(message, sig []byte) bool ` on `secp256k1.PublicKey`. Use `secp256k1test` package to test signed messages with both `sofia` and `charlotte` keys. This will mean that the validity of the signed data against this public key / key type can be verified.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by developerfred** [comment]
Test don't work on secp256k1
```
# command-line-arguments [command-line-arguments.test]
./public_test.go:41:31: undefined: charlottePublicKey
./public_test.go:43:4: undefined: charlottePublicKey
./public_test.go:49:24: undefined: charlottePublicKey
./public_test.go:51:4: undefined: charlottePublicKey
./public_test.go:59:4: undefined: charlottePublicKey
./public_test.go:67:4: undefined: charlottePublicKey
./public_test.go:89:16: undefined: PublicKeyFromHex
./public_test.go:122:20: undefined: ecdsaPrivateKeySofia
./public_test.go:123:21: undefined: ecdsaPrivateKeySofia
./public_test.go:143:16: undefined: ecdsaPrivateKeySofia
./public_test.go:143:16: too many errors
FAIL	command-line-arguments [build failed]
FAIL
```

**Comment by robdefeo** [comment]
@developerfred can you try running `go test github.com/mailchain/mailchain/crypto/secp256k1`

**Comment by robdefeo** [comment]
This can now be completed as #375 has been completed #373 provides a good example of how this can be implemented. Especially with the unit tests.

**Comment by Despire** [comment]
Is anybody working on this ? If now can I be assigned ? :)

**Comment by tboeckmann** [comment]
Thanks @Despire I have assigned you.

Checking with @developerfred in case any work was started, but I have seen anything.

**Comment by developerfred** [comment]
@Despire 

> Is anybody working on this ? If not, can I be assigned ? :)
You can do this feature, thanks for the reminder.

**Comment by Despire** [comment]
@developerfred seems like you already done pretty much all the work :) are you sure you don't want to finish it ?

**Comment by robdefeo** [comment]
closed by #416
